camel/agents/chat_agent.py

```diff
@@ -294,7 +294,8 @@ def step(
                 a boolean indicating whether the chat session has terminated,
                 and information about the chat session.
         """
-        messages = self.update_messages('user', input_message)
+        messages = self.update_messages(input_message.role_type.value,
```

Comment:
Since we had this bug in the code, please add to the test of `ai_society\role_playing.py` a check of the sequence of roles:
```
system
user
assistant
user
assistant
...
```
or whatever it must be. I expect added checks in `test/agents/test_role_playing.py`.

Reply:
Wait a second. I recall figuring out this complex behavior of changing the roles. I even put an explicit comment:
```
Its `role` field that specifies the role at backend may be either
            `user` or `assistant` but it will be set to `user` anyway since
            for the self agent any incoming message is external.
```

also see the doctoring above:
```
def submit_message(self, message: BaseMessage) -> None:
        r"""Submits the externally provided message as if it were an answer of
        the chat LLM from the backend. Currently, the choice of the critic is
        submitted with this method.
```

Please let me know if you think this is in the scope of this change or not. If not, make a separate bug ticket and a separate PR with the proper fix and the test, and revert the irrelevant changes in thin PR.

Reply:
Wait. I don't think this is a bug.